patches/chromium/add_streaming-schemes_registry.patch

```diff
@@ -0,0 +1,133 @@
+determination will cause the video/audio to fail playing.
+
+This patch adds a list of "streaming protocols" to the url SchemeRegistry in order to allow
+other protocols to register their streaming behavior. MultibufferDataSource::AssumeFullyBuffered()
+then refers to the registry so that it can correctly determine the data source's settings.
+
+diff --git a/media/blink/multibuffer_data_source.cc b/media/blink/multibuffer_data_source.cc
+index 0f6ae1fb8b4ff9f24ce3f407b7359e016fc6de5f..62c280d84643f26567bac51eadbfd78e8d5fd032 100644
+--- a/media/blink/multibuffer_data_source.cc
++++ b/media/blink/multibuffer_data_source.cc
+@@ -158,7 +158,7 @@ bool MultibufferDataSource::media_has_played() const {
+ 
+ bool MultibufferDataSource::AssumeFullyBuffered() const {
+   DCHECK(url_data_);
+-  return !url_data_->url().SchemeIsHTTPOrHTTPS();
++  return !url_data_->url().SchemeIsStreaming();
+ }
+ 
+ void MultibufferDataSource::SetReader(MultiBufferReader* reader) {
+diff --git a/url/gurl.cc b/url/gurl.cc
+index fb06480ed6e54e74c881dbe4f3aa4ea9780a8907..9ea2c86576512dccaeae4dd70614d88c22d01e1f 100644
+--- a/url/gurl.cc
++++ b/url/gurl.cc
+@@ -354,6 +354,10 @@ bool GURL::SchemeIsHTTPOrHTTPS() const {
+   return SchemeIs(url::kHttpScheme) || SchemeIs(url::kHttpsScheme);
+ }
+ 
++bool GURL::SchemeIsStreaming() const {
++  return is_valid_ && IsStreamingScheme(spec_.data(), parsed_.scheme);
++}
++
+ bool GURL::SchemeIsValidForReferrer() const {
+   return is_valid_ && IsReferrerScheme(spec_.data(), parsed_.scheme);
+ }
+diff --git a/url/gurl.h b/url/gurl.h
+index ad9f565586eb4d23f4508148f8f6192bb112ac86..d05d4791eca3a9782716648398abefefb5f5f716 100644
+--- a/url/gurl.h
++++ b/url/gurl.h
+@@ -229,6 +229,9 @@ class COMPONENT_EXPORT(URL) GURL {
+   // Returns true if the scheme is "http" or "https".
+   bool SchemeIsHTTPOrHTTPS() const;
+ 
++  // Returns true if the scheme is known to stream its response.
++  bool SchemeIsStreaming() const;
++
+   // Returns true if the scheme is valid for use as a referrer.
+   bool SchemeIsValidForReferrer() const;
+ 
+diff --git a/url/url_util.cc b/url/url_util.cc
+index 93a106328c33a5a330ff79ad9b9ca66b0a23a4d3..bf3ae674112bd3e7ed1c695806de2b95f513f20e 100644
+--- a/url/url_util.cc
++++ b/url/url_util.cc
+@@ -48,6 +48,12 @@ struct SchemeRegistry {
+       {kQuicTransportScheme, SCHEME_WITH_HOST_AND_PORT},
+   };
+ 
++  // Schemes that can stream their responses.
++  std::vector<std::string> streaming_schemes = {
++      kHttpsScheme,
++      kHttpScheme
++  };
++
+   // Schemes that are allowed for referrers.
+   std::vector<SchemeWithType> referrer_schemes = {
+       {kHttpsScheme, SCHEME_WITH_HOST_PORT_AND_USER_INFORMATION},
+@@ -551,6 +557,10 @@ void AddStandardScheme(const char* new_scheme, SchemeType type) {
+                       &GetSchemeRegistryWithoutLocking()->standard_schemes);
+ }
+ 
++void AddStreamingScheme(const char* new_scheme) {
```

Comment:
Argument `SchemeType` is not defined here, hence undefined symbols with overloaded functions.

Reply:
Aha, I'll give that a fix.